### PR TITLE
DOCS-8114 asm topic reorg for product focus

### PR DESCRIPTION
**Pull request by michaelcretzman**
DOCS-8114

### What does this PR do? What is the motivation?

Reorg of the ASM docs so that their onboarding info is more product-focused

### Merge instructions

Don't merge until I've reviewed with the PM.

- [ ] Please merge after reviewing

### CODE REVIEW

**Comment by aliciascott** [comment]
Hi @michaelcretzman I started reviewing this but then realized there is something wrong with the menu, a lot of things are still linking to `/security/application_security/enabling/` but that doesn't seem to exist anymore on the menu, for example should both:
  `/security/application_security/enabling/compatibility/` and 
`/security/application_security/threats/setup/compatibility/` exist? 

They look slightly different but there are links to both those pages from different areas, however, the menu doesn't exist for that location:

image

I "think" you want to get rid of any previous references and links to that old `enabling` directory since they appear to be replaced by new files, let me know if you can fix that up then I'll do another review, thanks!

**Comment by michaelcretzman** [comment]
@aliciascott I've removed the links to the old Enabling folder. I also had to update a doc to resolve a merge conflict that resulted from a PR checked in while my branch is out.

Preview:
https://docs-staging.datadoghq.com/mcretzman/DOCS-8114-asm-topic-reorg/security/application_security/

File: content/en/security/application_security/code_security/_index.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ further_reading:
 
 Datadog Code Security identifies code-level vulnerabilities in your services and provides actionable insights and recommended fixes.
 
-For a list of supported services, see [Library Compatibility Requirements][5].
+For a list of supported services, see the [Library Compatibility Requirements][5].
```
Comment (aliciascott) [inline content/en/security/application_security/code_security/_index.md:22]:
```suggestion
For a list of supported services, see the [Library Compatibility Requirements][5].
```

I think something is wrong with this page `/security/application_security/code_security/setup/compatibility/` 

It correctly only shows the 4 supported libraries:
JAVA | .NET | NODE.JS | PYTHON

But then when I click on one of those say, Java: it takes me to a page where all of the libraries are listed so It's kinda confusing, I would expect it to just take me to a page that only has those 4 libraries:

image

File: content/en/security/application_security/code_security/setup/compatibility/nginx.md
Comment (aliciascott) [inline content/en/security/application_security/code_security/setup/compatibility/nginx.md:1]: I don't think this page should be under Code Security as nginx is only supported for threats ?  (It seems to be in this folder but not on the menu) So I think you just need to remove it from `content/en/security/application_security/code_security/setup/compatibility/`

File: content/en/security/application_security/code_security/setup/compatibility/php.md
Comment (aliciascott) [inline content/en/security/application_security/code_security/setup/compatibility/php.md:1]: Same with this one, there is no php for Code Security

File: content/en/security/application_security/code_security/setup/compatibility/ruby.md
Comment (aliciascott) [inline content/en/security/application_security/code_security/setup/compatibility/ruby.md:1]: Also ruby needs removed from the code security folder

File: content/en/security/application_security/code_security/setup/compatibility/serverless.md
Comment (aliciascott) [inline content/en/security/application_security/code_security/setup/compatibility/serverless.md:1]: Same with serverless

File: content/en/security/application_security/software_composition_analysis/setup/compatibility/serverless.md
Comment (aliciascott) [inline content/en/security/application_security/software_composition_analysis/setup/compatibility/serverless.md:1]: I think this file needs removed too as we don't have Serverless for SCA

File: content/en/security/application_security/code_security/setup/_index.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+2. **Datadog APM Configuration:** Datadog APM is configured for your application or service, and traces are being received by Datadog.
+3. **Supported Tracing Library:** The Datadog Tracing Library used by your application or service supports Software Composition Analysis capabilities for the language of your application or service. For more details, refer to the [Library Compatibility][1] page.
+
+## Code Security Enablement Types
+
+There are two main approaches to enable Code Security on your tracing libraries: Single-Step Instrumentation and Datadog Tracing Libraries.
+
+### Single-Step Instrumentation
+
+Run a one-line install command to install the Datadog Agent, and enable Code Security with [Single-Step Instrumentation][2].
```
Comment (aliciascott) [inline content/en/security/application_security/code_security/setup/_index.md:32]:
```suggestion
Run a one-line install command to install the Datadog Agent, and enable Code Security with [Single-Step Instrumentation][2].
```
```suggestion
Run a one-line install command to install the Datadog Agent, and enable Code Security with [Single-Step Instrumentation][2].
```
this link just links back to the same page, where should it go?

**Review by aliciascott — CHANGES_REQUESTED** [review]
Going to just review this in phases, I got through 40 files and some comments here where some files and links need removed/replaced, let me know when those are fixed then I'll review the rest :) Also be sure to add alias to the front matter of the new page pointing to the old location. 
https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/2174550131/Moving+Docs+Pages

File: content/en/security/application_security/code_security/setup/_index.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+{{< partial name="security-platform/appsec-languages-code-security.html" >}}</br>
+
+## Further reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+
+
+[1]: /security/application_security/enabling/compatibility/
```
[the remark this answers is not in the document]
Comment (michaelcretzman) [reply content/en/security/application_security/code_security/setup/_index.md]: this is fixed in my latest push.

File: content/en/security/application_security/code_security/_index.md (hunk 1 above)
Comment (michaelcretzman) [reply content/en/security/application_security/code_security/_index.md:22]: I forgot to remove the other language pages and so Hugo was displaying them at the top.
I will commit a fix.

Comment (michaelcretzman) [reply content/en/security/application_security/code_security/_index.md:22]: fix committed.

File: content/en/security/application_security/code_security/setup/compatibility/nginx.md
Comment (michaelcretzman) [reply content/en/security/application_security/code_security/setup/compatibility/nginx.md:1]: done

File: content/en/security/application_security/code_security/setup/compatibility/php.md
Comment (michaelcretzman) [reply content/en/security/application_security/code_security/setup/compatibility/php.md:1]: done

File: content/en/security/application_security/code_security/setup/compatibility/ruby.md
Comment (michaelcretzman) [reply content/en/security/application_security/code_security/setup/compatibility/ruby.md:1]: done

File: content/en/security/application_security/code_security/setup/compatibility/serverless.md
Comment (michaelcretzman) [reply content/en/security/application_security/code_security/setup/compatibility/serverless.md:1]: done

File: content/en/security/application_security/software_composition_analysis/setup/compatibility/serverless.md
Comment (michaelcretzman) [reply content/en/security/application_security/software_composition_analysis/setup/compatibility/serverless.md:1]: done. You will see it in my next commit.

File: content/en/security/application_security/code_security/setup/_index.md (hunk 2 above)
Comment (michaelcretzman) [reply content/en/security/application_security/code_security/setup/_index.md]: fixed. I incorporated the entire file into the main setup. So it's a one-pager now.
You will see it in my next commit.

**Review by aliciascott — APPROVED** [review]
Hi @michaelcretzman this looks great, and it was a huge lift! Feel free to merge when good for you, I don't see any other issues :)

**Pull request merged**